types/leaflet/index.d.ts

```diff
@@ -424,11 +424,11 @@ declare namespace L {
         getTooltip(): Tooltip | undefined;
 
         // Extension methods
-        onAdd?: (map: Map) => this;
-        onRemove?: (map: Map) => this;
-        getEvents?: () => {[name: string]: (event: Event) => void};
-        getAttribution?: () => string | null;
-        beforeAdd?: (map: Map) => this;
+        onAdd?(map: Map): this;
```

Comment:
These are not optional. When I extend `L.Marker` I should be able `super.onAdd` in my implementation